darts/models/forecasting/regression_model.py

```diff
@@ -282,65 +282,94 @@ def _get_last_prediction_time(self, series, forecast_horizon, overlap_end):
-                (future_covariate, self.lags_future_covariates),
+                (past_covariates[idx].pd_dataframe(), self.lags_past_covariates)
+                if past_covariates
+                else (None, None),
+                (
+                    future_covariates[idx].pd_dataframe(),
+                    (
+                        self.lags_historical_covariates
+                        if self.lags_historical_covariates
+                        else []
+                    )
+                    + (
+                        self.lags_future_covariates
+                        if self.lags_future_covariates
+                        else []
+                    ),
+                )
+                if future_covariates
+                else (None, None),
             ]
-            for covariate, lags in covariates:
-                if lags is not None:
-                    row.append(covariate[lags].reshape(1, -1))
 
-            training_samples.append(np.concatenate(row, axis=1))
-            training_labels.append(future_target[0].reshape(1, -1))
+            df_X = []
+            df_y = target_ts.pd_dataframe()
 
-        training_samples = np.concatenate(training_samples, axis=0)
-        training_labels = np.concatenate(training_labels, axis=0)
-        return training_samples, training_labels
+            # target lags
+            if self.lags:
+                for lag in self.lags:
+                    df_X.append(df_y.shift(-lag))
 
-    def _create_lagged_data(
-        self, target_series, past_covariates, future_covariates, max_samples_per_ts
-    ):
-        """
-        Helper function that creates training/validation matrices (X and y as required in sklearn), given series and
-        max_samples_per_ts.
-        """
+            # covariate lags
+            for df_cov, lags in covariates:
+                if lags:
+                    for lag in lags:
+                        df_X.append(df_cov.shift(-lag))
 
-        # setting proper input and output chunk length. We need to find all necessary lags in the dataset. Thus, we
-        # need an input chunk length that contains the oldest past lag (min_lag), and we need to find the most future
-        # lag in the future_covariates as well. In case no future covariate are required, we still need an output
-        # chunk length of 1 for having the prediction target value.
-        training_dataset = MixedCovariatesSequentialDataset(
-            target_series=target_series,
-            past_covariates=past_covariates,
-            future_covariates=future_covariates,
-            input_chunk_length=max(0, -self.min_lag),
-            output_chunk_length=max(
-                1, self.max_lag + 1
-            ),  # max lag + 1 since we need to access that index
-            max_samples_per_ts=max_samples_per_ts,
-        )
+            # combine lags
+            df_X = pd.concat(df_X, axis=1)
```

Comment:
This is neat 👌

Reply:
Fortunately, I could let pd.concat do the task of aligning the different series with each other 😄